lib/geo_postgis/geometry.ex

```diff
@@ -57,6 +57,22 @@ if Code.ensure_loaded?(Ecto.Type) do
       GeometryCollection
     ]
 
+    @type t ::
+            Point.t()
+            | PointZ.t()
+            | PointM.t()
+            | PointZM.t()
+            | LineString.t()
+            | LineStringZ.t()
+            | Polygon.t()
+            | PolygonZ.t()
+            | MultiPoint.t()
+            | MultiPointZ.t()
+            | MultiLineString.t()
+            | MultiLineStringZ.t()
+            | MultiPolygon.t()
+            | MultiPolygonZ.t()
```

Comment:
We could use some macro to generate this..

something like
```elixir
@type t :: unquote(Enum.reduce(@geometries, fn g, acc -> {:|, [], [g, acc]} end))
```
 but I'm not sure if it's better..

Reply:
Mmm, right... I do think we should build the type automatically from the `@geometries` list, just so that the long-term maintenance doesn't depend on remembering to add new types in both places. We could potentially wrap the AST generation in a macro [like this](https://github.com/felt/union_typespec/blob/main/lib/union_typespec.ex), but I'd be comfortable with the above inline `unquote` with a comment explaining that it's producing the union of all the geometry types.

Reply:
hey! actually this is already declared as [Geo.geometry()](https://github.com/felt/geo/blob/c85ebc0bbdd9df1184e36ecb3d9538a1f392642d/lib/geo.ex#L6-L21) all we need is just to refer to it!